From 3712ca572d659dc6033899ea1cfc6ed7ba306bf4 Mon Sep 17 00:00:00 2001
From: Krishna Kanth Reddy <krish.reddy@samsung.com>
Date: Wed, 14 Jul 2021 13:03:31 +0530
Subject: [PATCH 5/7] ARM64: Cross-Compilation Support

Modified the Configuration file to use the latest ARM Cross-Compiler.

Fixed the linker errors for the undefined references to the APIs
isal_deflate_init, isal_deflate, isal_inflate_init, isal_inflate,
isal_inflate_stateless, isal_deflate_stateless,
isal_deflate_set_hufftables in the case of ARM Cross-Compilation.

Change-Id: I0ba89e5640760276646d6b9211585ad116ebf446
Signed-off-by: Krishna Kanth Reddy <krish.reddy@samsung.com>
Signed-off-by: Krzysztof Karas <krzysztof.karas@intel.com>
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/8778 (spdk-21.05)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/9218 (spdk-21.08)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/10450 (spdk-21.11)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/12500
(spdk-21.11.1)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/14311
Tested-by: SPDK CI Jenkins <sys_sgci@intel.com>
Reviewed-by: Konrad Sztyber <konrad.sztyber@intel.com>
Reviewed-by: Ben Walker <benjamin.walker@intel.com>
---
 config/arm/arm64_armv8_linux_gcc  | 10 +++++-----
 drivers/compress/isal/meson.build |  4 ++++
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/config/arm/arm64_armv8_linux_gcc b/config/arm/arm64_armv8_linux_gcc
index 5c32f6b9ca..4f56798fae 100644
--- a/config/arm/arm64_armv8_linux_gcc
+++ b/config/arm/arm64_armv8_linux_gcc
@@ -1,9 +1,9 @@
 [binaries]
-c = 'aarch64-linux-gnu-gcc'
-cpp = 'aarch64-linux-gnu-g++'
-ar = 'aarch64-linux-gnu-gcc-ar'
-strip = 'aarch64-linux-gnu-strip'
-pkgconfig = 'aarch64-linux-gnu-pkg-config'
+c = 'aarch64-none-linux-gnu-gcc'
+cpp = 'aarch64-none-linux-gnu-g++'
+ar = 'aarch64-none-linux-gnu-gcc-ar'
+strip = 'aarch64-none-linux-gnu-strip'
+pkgconfig = 'aarch64-none-linux-gnu-pkg-config'
 pcap-config = ''
 
 [host_machine]
diff --git a/drivers/compress/isal/meson.build b/drivers/compress/isal/meson.build
index 03294c71e7..5daf24d113 100644
--- a/drivers/compress/isal/meson.build
+++ b/drivers/compress/isal/meson.build
@@ -5,6 +5,10 @@ dep = dependency('libisal', required: false, method: 'pkg-config')
 if not dep.found()
     build = true
     reason = 'missing dependency, "libisal"'
+    isal_dep = cc.find_library('libisal', required: false)
+    if isal_dep.found()
+        ext_deps += isal_dep
+    endif
 endif
 
 deps += 'bus_vdev'
-- 
2.31.1

